app/module/configurator.go

```diff
@@ -0,0 +1,93 @@
+package module
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/grpc"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+type configurator struct {
+	cdc         codec.Codec
+	msgServer   grpc.Server
+	queryServer grpc.Server
+
+	// migrations is a map of moduleName -> fromVersion -> migration script handler
+	migrations map[string]map[uint64]module.MigrationHandler
+}
+
+// NewConfigurator returns a new Configurator instance
+func NewConfigurator(cdc codec.Codec, msgServer grpc.Server, queryServer grpc.Server) module.Configurator {
+	return configurator{
+		cdc:         cdc,
+		msgServer:   msgServer,
+		queryServer: queryServer,
+		migrations:  map[string]map[uint64]module.MigrationHandler{},
+	}
+}
+
+var _ module.Configurator = configurator{}
+
+// MsgServer implements the Configurator.MsgServer method
+func (c configurator) MsgServer() grpc.Server {
+	return c.msgServer
+}
+
+// QueryServer implements the Configurator.QueryServer method
+func (c configurator) QueryServer() grpc.Server {
+	return c.queryServer
+}
+
+// RegisterMigration implements the Configurator.RegisterMigration method
+func (c configurator) RegisterMigration(moduleName string, fromVersion uint64, handler module.MigrationHandler) error {
+	if fromVersion == 0 {
+		return sdkerrors.ErrInvalidVersion.Wrap("module migration versions should start at 1")
```

Comment:
does this mean no migrations can occur from v0 -> v1? Does other code already assume version starts at v1?

Reply:
Correct. Yes we started on v1 and there are assumptions that v0 doesn't exist and implies that the version has forgotten to be set

Reply:
thanks! 

[tangent] Is there already an assertion somewhere that fails on chain start up if a chain is initialized with v0? Asking because I haven't seen an assertion like that but I expect it exists _somewhere_.

Reply:
Yes I believe I've added it. I'll double check